lib/persisted-model.js

```diff
@@ -1060,19 +1073,25 @@ PersistedModel.bulkUpdate = function(updates, callback) {
   var Model = this;
   var idName = this.dataSource.idName(this.modelName);
   var Change = this.getChangeModel();
+  var conflicts = [];
 
   updates.forEach(function(update) {
+    var id = update.change.modelId;
     switch (update.type) {
       case Change.UPDATE:
+        tasks.push(function(cb) {
+          applyUpdate(Model, id, update.data, update.change, conflicts, cb);
+        });
+        break;
+
       case Change.CREATE:
-        // var model = new Model(update.data);
-        // tasks.push(model.save.bind(model));
         tasks.push(function(cb) {
-          var model = new Model(update.data);
-          model.save(cb);
+          applyCreate(Model, id, update.data, update.change, conflicts, cb);
         });
         break;
       case Change.DELETE:
+        // NOTE(bajtos) Delete does not care whether the model has
+        // the same revision as expected, we are going to delete it anyway
```

Comment:
The fix for this requires https://github.com/strongloop/loopback-datasource-juggler/pull/511 and possibly implementation in all connectors.